### ISSUE TITLE
Assumptions about self signed state

### ISSUE THREAD

**Issue by mttcpr** [wbond/asn1crypto#73]
I noticed you are currently using key ids to make assumptions about self signed state in asn1crypto/x509. 

https://github.com/wbond/asn1crypto/blob/16f4ce0c4a809486621ee8333ed5250d2e509818/asn1crypto/x509.py#L2518

I would consider replacing that 'yes' with 'maybe' because (1) key IDs are not always correct, and, (2) users would likely assume that self_signed == True means the signature was verified. It'd be safer to bite the CPU bullet and check the signature. 

Note that this suggestion is loosely coupled to https://github.com/wbond/oscrypto/issues/21

If you change those yeses to maybes without broadening the oscrypto SignatureError exception cases or otherwise handling those mismatch errors, this code in CertValidator will emit exceptions much more regularly:

```
    # In the case of "maybe", we have to check the signature
    signature_algo = cert['signature_algorithm'].signature_algo
    hash_algo = cert['signature_algorithm'].hash_algo

    if signature_algo == 'rsassa_pkcs1v15':
        verify_func = asymmetric.rsa_pkcs1v15_verify
    elif signature_algo == 'dsa':
        verify_func = asymmetric.dsa_verify
    elif signature_algo == 'ecdsa':
        verify_func = asymmetric.ecdsa_verify
    else:
        raise PathValidationError(pretty_message(
            '''
            Unable to verify the signature of the certificate since it uses
            the unsupported algorithm %s
            ''',
            signature_algo
        ))

    try:
        key_object = asymmetric.load_certificate(cert)
        verify_func(key_object, cert['signature_value'].native, cert['tbs_certificate'].dump(), hash_algo)
        return True

    except (oscrypto.errors.SignatureError):
        return False

```

**Comment by joernheissler** [comment]
To quote from rfc 5280 section §3.2:
`Self-issued certificates are CA certificates in which the issuer and subject are the same entity. Self-issued certificates are generated to support changes in policy or operations. Self-signed certificates are self-issued certificates where the digital signature may be verified by the public key bound into the certificate. Self-signed certificates are used to convey a public key for use to begin certification paths.`

So in my opinion, the property could be renamed self_issued and made a bool. In absense of the Authority + Subject Key identifiers, the non-extension issuer+subject fields could be compared.

self_signed, with actual verification of the signature, is probably out of scope for asn1crypto. It's an ASN.1 library without any dependencies. So there is no access to crypto libs that would be required to verify the signature.

**Comment by mttcpr** [comment]
The self_issued property is there already and seems fine to me. I agree to an extent that self_signed is out of scope, but I also think it's a nice property for the cert to be able to carry with it, which I imagine is why Will put it there. Setting it to maybe instead of yes will cause the code in CertValidator to do the crypto (which is exactly where we should be doing the verification) instead of assuming the signature is valid without checking. There is still no crypto being performed in asn1crypto in any case - it's up to the consumer of the cert to do what it will with the 'maybe'.

**Comment by wbond** [comment]
For backwards compatibility, it probably makes sense that `self_signed` can only be `no` or `maybe`. Just never return the `yes` value any longer. That way we don't break the API, but also aren't misleading people.

**Comment by wbond** [comment]
This should be resolved by 7647163
